### DISCUSSION TITLE
Order Items List for all Orders

### DISCUSSION THREAD

**Discussion by LuBla98** [saleweaver/python-amazon-sp-api#144] [Q&A] [answered]
Hi,
i try to get an Order Items List for all of our Orders per Day. But for "get_order_item"s i have to give the Parameter "order_id". If i make a simple API request for every Order-ID from the day, i get over the rate limit. Is there anyone who has the same problem and maybe give me an idea?

**Comment by saleweaver** [comment, marked as answer]
The recommended approach is to use reports instead. 
Do you really have tens of thousands of orders per day?

**Reply by LuBla98** [reply]
Ok, thats an idea. But everytime i try to create a report i get the Message:
`[{'code': 'Unauthorized', 'message': 'Access to the resource is forbidden', 'details': ''}]`

I tried it with this code:
`Reports(credentials=credentials, marketplace=Marketplaces.DE).create_report(reportType='GET_FLAT_FILE_OPEN_LISTINGS_DATA')`

**Reply by saleweaver** [reply]
This error is unrelated to rate limiting, you'll need to add the permission to your app (I think it's Inventory Management or so). 
Make sure to regenerate the refresh token after you added the permission. 
You should be able to get on of the order reports if you had access to the orders api.
